src/components/shared/Navbar.tsx

```diff
@@ -33,10 +115,22 @@ const Navbar: React.FC = () => {
     getRoles();
   }, [user?.uid]);
 
+  React.useEffect(() => {
```

Comment:
I don't think you need a useEffect for this or a state variable for the hexathon, I think you can probably just make the request once per component render and then just return the loading screen if it hasn't loaded. We do this commonly in a lot of other parts of this repo so just take a look there for reference